pkg/controller/machinedeployment/sync.go

```diff
@@ -128,6 +128,11 @@ func (r *ReconcileMachineDeployment) getNewMachineSet(d *clusterv1alpha1.Machine
 	// Add machineTemplateHash label to selector.
 	newMSSelector := dutil.CloneSelectorAndAddLabel(&d.Spec.Selector, dutil.DefaultMachineDeploymentUniqueLabelKey, machineTemplateSpecHash)
 
+	minReadySeconds := int32(0)
```

Comment:
Would it make more sense to add a `PopulateDefaultsMachineSet` function to set this default value so that we also get it set to 0 if you create a machine set directly? In which case you could just do

`MinReadySeconds: d.Spec.MinReadySeconds`

below?

Reply:
Yes, although oddly d is a MachineDeployment, so I should have already done this...  I guess I'll revert this bit and then we can see what breaks; it might just be an artifact of the order I fixed things (i.e. might be unnecessary), or it might be that I've missed a call to PopulateDefaultsMachineDeployment